### PR TITLE
RavenDB-15078 Add time-series configuration to export options

### PR DESCRIPTION
**Pull request by ml054**

### CODE REVIEW

File: src/Raven.Client/Documents/Smuggler/DatabaseItemType.cs [hunk 1]
```diff
@@ -50,6 +50,7 @@ public enum DatabaseRecordItemType
         Sorters = 1 << 12,
         SinkPullReplications = 1 << 13,
         HubPullReplications = 1 << 14,
-        TimeSeries = 1 << 15
+        TimeSeries = 1 << 15,
+        DocumentCompression = 1 << 16
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/DatabaseItemType.cs]: DocumentsCompression

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs [hunk 2]
```diff
@@ -228,6 +228,8 @@ public class DatabaseRecordProgress : Counts
             public bool ConflictSolverConfigUpdated { get; set; }
 
             public bool TimeSeriesConfigurationUpdated { get; set; }
+            
+            public bool DocumentCompressionConfigurationUpdated { get; set;  }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: DocumentsCompression

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs [hunk 3]
```diff
@@ -338,6 +343,12 @@ public override string ToString()
                 if (ClientConfigurationUpdated)
                     sb.AppendLine("- Client");
 
+                if (TimeSeriesConfigurationUpdated)
+                    sb.AppendLine("- Time Series");
+
+                if (DocumentCompressionConfigurationUpdated)
+                    sb.AppendLine("- Document Compression");
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: Documents Compression

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs [hunk 4]
```diff
@@ -759,6 +759,33 @@ public void WriteDatabaseRecord(DatabaseRecord databaseRecord, SmugglerProgressB
...
+                        }
+
+                        if (collectionsToAdd.Count > 0)
+                        {
+                            databaseRecord.DocumentsCompression.Collections = collectionsToAdd.Concat(currentDatabaseRecord.DocumentsCompression.Collections).ToArray();
+                        }
+                    }
+                    
+                    if (_log.IsInfoEnabled)
+                        _log.Info("Configuring document compression from smuggler");
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: documents compression

File: src/Raven.Server/Smuggler/Documents/StreamDestination.cs [hunk 5]
```diff
@@ -562,6 +569,16 @@ private void WriteTimeSeries(TimeSeriesConfiguration timeSeries)
                 _context.Write(_writer, timeSeries.ToJson());
             }
 
+            private void WriteDocumentCompression(DocumentsCompressionConfiguration compressionConfiguration)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/StreamDestination.cs]: DocumentsCompression

File: src/Raven.Server/Smuggler/Documents/StreamSource.cs [hunk 6]
```diff
@@ -152,6 +152,20 @@ public DatabaseRecord GetDatabaseRecord()
...
+                    && documentCompression != null)
+                {
+                    try
+                    {
+                        databaseRecord.DocumentsCompression = JsonDeserializationCluster.DocumentsCompressionConfiguration(documentCompression);
+                    }
+                    catch (Exception e)
+                    {
+                        if (_log.IsInfoEnabled)
+                            _log.Info("Wasn't able to import the document compression configuration from smuggler file. Skipping.", e);
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/StreamSource.cs]: documents compression

File: src/Raven.Studio/typescript/models/database/tasks/smugglerDatabaseRecord.ts [hunk 7]
```diff
@@ -105,6 +107,12 @@ class smugglerDatabaseRecord {
         if (this.includeHubPullReplications()) {
             result.push("HubPullReplications");
         }
+        if (this.includeDocumentCompression()) {
+            result.push("DocumentCompression");
```
Comment (ppekrol) [inline src/Raven.Studio/typescript/models/database/tasks/smugglerDatabaseRecord.ts]: DocumentsCompression

File: src/Raven.Client/Documents/Smuggler/DatabaseItemType.cs (hunk 1 above)
Comment (ml054) [reply src/Raven.Client/Documents/Smuggler/DatabaseItemType.cs]: or maybe compression?

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs (hunk 3 above)
Comment (ml054) [reply src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: in studio we have Document Compression I matched it

Comment (ml054) [reply src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: image

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs (hunk 2 above)
Comment (ml054) [reply src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: done

File: src/Raven.Client/Documents/Smuggler/DatabaseItemType.cs (hunk 1 above)
Comment (ml054) [reply src/Raven.Client/Documents/Smuggler/DatabaseItemType.cs]: done

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs (hunk 3 above)
Comment (ml054) [reply src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: done

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs [hunk 8]
```diff
@@ -759,6 +759,33 @@ public void WriteDatabaseRecord(DatabaseRecord databaseRecord, SmugglerProgressB
                     progress.TimeSeriesConfigurationUpdated = true;
                 }
 
+                if (databaseRecord?.DocumentsCompression != null)
+                {
+                    if (currentDatabaseRecord?.DocumentsCompression != null)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: wouldn't it be more safer to do?
```
currentDatabaseRecord?.DocumentsCompression.Collections?.Count > 0
```

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs (hunk 8 above)
Comment (ml054) [reply src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: done

**Pull request merged**